ci/ci

```diff
@@ -353,6 +353,9 @@ if __name__ == "__main__":
             write_to_file(env_file, "CONTAINER_PLATFORM="+val)
             if args.build != None:
                 delete_older_image(val)
+                if val.find("ci_pbs") != -1:
+                    val = val[:-7].replace('_', ':')
```

Comment:
I understand, we are replacing the "_" right before "ci_pbs", however, adding a comment as to why -7 is used, will be good.
should we also confirm "ci_pbs" is indeed at the end of val?

Reply:
also what if val is like "xyz_ci_pbs:v1" (aka image name with tag id)?

Reply:
The image name is generated by the script itself, so it will have '_ci_pbs' at the end only. The user is supposed to give an image recoginized in the docker hub. For eg if the image is 'centos:7' then the saved image from ci will have the name 'centos_7_ci_pbs' which will be converted back to 'centos:7' after the change. There is no scenario where the image saved in the .env file is like "xyz_ci_pbs:v1", it always is "xyz_ci_pbs" latest version.

Reply:
what if I have a custom image generated from centos:7 and I named it xyz_ci_pbs:v1?
You are using val in this if block which is passed to --start an argument. So user can pass any image name in --start right?

Reply:
Yes in that case the image saved will be 'xyz_ci_pbs_v1_ci_pbs', the script will add a suffix to whatever image you provide.

Reply:
I don't think so, as per code, it will be 'xyz_ci_pbs:v1_ci_pbs' because you expect last 7 char should be "_ci_pbs" but in above case, last 7 char is '_pbs:v1'

Reply:
That is an optional condition. If it dosen't get the last characters as '_ci_pbs' it goes ahead with the given image and when the first time build completes, it will then commit it as 'xyz_ci_pbs_v1_ci_pbs'.

Reply:
* Modified the condition to make sure only the end of an image name contains '_ci_pbs'.  
* Also when converting back to old name, only change the last "_" to ":" as ":" is separator for name and version in docker images, and only one should exist.

I tested the  condition which @hirenvadalia suggested using an image 'ubuntu_ci_pbs:v1' and  posting the result below:
```
sanidhya@ubuntu:~/git_chng/git_iss/pbspro/ci$ ./ci --start='ubuntu_ci_pbs:v1'
13:26:16 ---> No running service found
13:26:16 ---> Attempting to start container
Removing network ci_ci
WARNING: Network ci_ci not found.
Creating network "ci_ci" with driver "bridge"
Creating ci_pbs-build_1 ... done
13:26:18 ---> Waiting for container build to complete 
13:26:18 ---> Build logs can be found in /home/sanidhya/git_chng/git_iss/pbspro/ci/logs/build
13:26:48 ---> Current status ...... Get:37 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 git amd64 1:2.17.1-1ubuntu0.5 [3912 kB]
...
13:27:48 ---> Current status ...... Unpacking libllvm8:amd64 (1:8-3~ubuntu18.04.2) over (1:8-3~ubuntu18.04.1) ...

13:28:18 ---> Current status ...... ### First time build is complete ###

13:28:18 ---> build is now complete
13:28:19 ---> docker commit 2e1773a1694d ubuntu_ci_pbs_v1_ci_pbs:latest
13:28:21 --->  done start container 
sanidhya@ubuntu:~/git_chng/git_iss/pbspro/ci$ ./ci --delete
Stopping ci_pbs-build_1 ... done
Removing ci_pbs-build_1 ... done
Removing network ci_ci
13:29:18 ---> done delete container and services
sanidhya@ubuntu:~/git_chng/git_iss/pbspro/ci$ cat .env 
CONTAINER_PLATFORM=ubuntu_ci_pbs:v1sanidhya@ubuntu:~/git_chng/git_iss/pbspro/ci$ ./ci --start 
13:29:26 ---> No running service found
13:29:26 ---> Attempting to start container
Removing network ci_ci
WARNING: Network ci_ci not found.
Creating network "ci_ci" with driver "bridge"
Creating ci_pbs-build_1 ... done
13:29:28 --->  done start container 
sanidhya@ubuntu:~/git_chng/git_iss/pbspro/ci$ ./ci --delete
Stopping ci_pbs-build_1 ... done
Removing ci_pbs-build_1 ... done
Removing network ci_ci
13:29:41 ---> done delete container and services
sanidhya@ubuntu:~/git_chng/git_iss/pbspro/ci$ cat .env 
CONTAINER_PLATFORM=ubuntu_ci_pbs_v1_ci_pbssanidhya@ubuntu:~/git_chng/git_iss/pbspro/ci$ ./ci --start --build
13:29:51 ---> The following untagged images will be removed -> 393d44ab79d2 
13:29:52 ---> No running service found
13:29:52 ---> Attempting to start container
Removing network ci_ci
WARNING: Network ci_ci not found.
Creating network "ci_ci" with driver "bridge"
Creating ci_pbs-build_1 ... done
13:29:54 ---> Waiting for container build to complete 
13:29:54 ---> Build logs can be found in /home/sanidhya/git_chng/git_iss/pbspro/ci/logs/build
13:30:24 ---> Current status ...... Get:15 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libpython3.6-dev amd64 3.6.9-1~18.04 [44.8 MB]
...
13:33:54 ---> Current status ...... ### First time build is complete ###

13:33:54 ---> build is now complete
13:33:54 ---> docker commit 3ab7b1aecfa6 ubuntu_ci_pbs_v1_ci_pbs:latest
13:33:57 --->  done start container 
sanidhya@ubuntu:~/git_chng/git_iss/pbspro/ci$ cat .env 
CONTAINER_PLATFORM=ubuntu_ci_pbs:v1sanidhya@ubuntu:~/git_chng/git_iss/pbspro/ci$ docker images | grep ubuntu_ci_pbs_v1_
ubuntu_ci_pbs_v1_ci_pbs   latest              098f5820daf6        2 minutes ago       1.08GB
```